### ISSUE TITLE
Skin Composer disallows special characters when naming resources

### ISSUE THREAD

**Issue by misterbander** [raeleus/skin-composer#123]
When naming resources (fonts, colors, drawables etc.), Skin Composer disallows the use of special characters (@, #, !, spaces etc.), even though libGDX's skin has no restrictions on resource naming.

![Screenshot_20220712_115316](https://user-images.githubusercontent.com/61056749/178408930-10716e6f-66ef-4e30-aba6-d858dc2f1d95.png)

Tested on version 51

**Comment by raeleus** [comment]
Yes, this is a limitation I added myself. In retrospect, it was an unnecessary, arbitrary limitation. I'll have to fix this in the future.

**Comment by tommyettinger** [comment]
However, libGDX atlases do have some bizarre restrictions! Unless you're using some self-written alternate atlas loading code, atlases are loaded with the system encoding. I had to get around this mess in TextraTypist by duplicating the asset loading code and changing one line where it loads a file without specifying UTF-8, which, yeah, it should. I pointed this out to the author of TextureAtlas, and after much fuss he tried to fix it but didn't. So I gave up and use my custom solution to this day.
